pennylane/transforms/qcut.py

```diff
@@ -939,6 +940,91 @@ def qcut_processing_fn_mc(
     return qml.math.convert_like(np.mean(expvals), res0)
 
 
+@batch_transform
+def cut_circuit_mc(
+    tape: QuantumTape,
+    shots: int,
+    device_wires: Optional[Wires] = None,
+    max_depth: int = 1,
+    classical_processing_fn: callable = None,
+) -> Tuple[Tuple[QuantumTape], Callable]:
+    """
+    TODO: Docstring
+    """
+
+    g = tape_to_graph(tape)
+    replace_wire_cut_nodes(g)
+    fragments, communication_graph = fragment_graph(g)
+    fragment_tapes = [graph_to_tape(f) for f in fragments]
+    fragment_tapes = [remap_tape_wires(t, device_wires) for t in fragment_tapes]
+
+    configurations, settings = expand_fragment_tapes_mc(
+        fragment_tapes, communication_graph, shots=shots
+    )
+
+    tapes = tuple(tape for c in configurations for tape in c)
+
+    if classical_processing_fn:
+        return tapes, partial(
+            qcut_processing_fn_mc,
+            communication_graph=communication_graph,
+            settings=settings,
+            shots=shots,
+            classical_processing_fn=classical_processing_fn,
+        )
+    else:
+        return tapes, partial(
+            qcut_processing_fn_sample, communication_graph=communication_graph, shots=shots
+        )
+
+
+@cut_circuit_mc.custom_qnode_wrapper
+def qnode_execution_wrapper(self, qnode, targs, tkwargs):
+    """Here, we overwrite the QNode execution wrapper in order
+    to replace execution variables"""
+
+    transform_max_diff = tkwargs.pop("max_diff", None)
+
+    if "shots" in inspect.signature(qnode.func).parameters:
+        raise ValueError(
+            "Detected 'shots' as an argument of the quantum function to transform. "
+            "The 'shots' argument name is reserved for overriding the number of shots "
+            "taken by the device."
+        )
+
+    def _wrapper(*args, **kwargs):
+        qnode.construct(args, kwargs)
```

Comment:
Within the wrapper, we need to work out the `shots` value that we'll pass to `self.construct`.

We need to:
- Check this: `shots = kwargs.pop("shots", False)`
- Do `shots = shots or device.shots`
- Set `tkwargs["device_wires"] = shots` (before calling `self.construct`)

Reply:
Thanks for the suggestion, I'm a bit confused with this. 

For the `test_cut_circuit_mc_expval` test below when I check `kwargs` here it's an empty `{}`, we don't have a `device` accessible but we do have `qnode.device` however `qnode.device.shots = None` so then `shots = shots or device.shots` becomes equivalent to `shots = False or None`.

I'm also not sure why we want to override the device wires with he number of shots in the transform keyword args? 

The logic without these steps gives a fairly accurate estimate of the expectation value in the tests for `shots = 10000`. Given that this occurs, I'm definitely missing something regarding these extra steps for the shots :thinking:

Reply:
```suggestion
        shots = kwargs.pop("shots", False)
        shots = shots or qnode.device.shots
        tkwargs["shots"] = shots
        
        qnode.construct(args, kwargs)
```
This is what I mean, hopefully this is clearer.

This allows us the following options:
1. (standard): since the user is sampling, they specify a `shots` value when loading their device. This is then visible in `qnode.device.shots`
2. (variant): in PL, you can optionally override the shots value of a device when evaluating a QNode by calling it with the `shots=100` argument. In this case, `kwargs.pop("shots", False)` would evaluate to `100`.
3. (specific to this transform): you can specify the `shots` value in the transform, without specifying in the device or when calling the QNode. I think it's ok that we support this (and this is the eventuality that you've done so far in the tests), but I think supporting option 1 and 2 will be more familiar to PL users.